propagation-stackdriver/src/main/java/zipkin2/propagation/stackdriver/CompositeExtractor.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright 2016-2018 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.propagation.stackdriver;
+
+import brave.propagation.TraceContext;
+import brave.propagation.TraceContextOrSamplingFlags;
+
+public class CompositeExtractor<C> implements TraceContext.Extractor<C> {
+
+	private TraceContext.Extractor<C>[] extractors;
```

Comment:
final version would hide the ctor final the field and use a factory method instead of ctor. inside that it would guard against null (not guava just throw NPE) or empty (length zero as IAE) and copy the input array.

Reply:
Should I be going for something other than arrays, given they don't support generics?